sklearn/metrics/scorer.py

```diff
@@ -279,6 +289,9 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
         For example ``average_precision`` or the area under the roc curve
         can not be computed using discrete predictions alone.
 
+    name : str, default=None
+        The name for the scorer. If None, the score_func.__name__ is used.
```

Comment:
Should we maybe have `None` default warn if `None` and set to `'custom'` and after 2 release make it non-default? Is this useful or am I digressing again?